.github/workflows/modem__build-host-tests.yml

```diff
@@ -0,0 +1,173 @@
+name: "esp-modem: build/host-tests"
```

Comment:
As we are merging host test and coverage tests into one file, I would suggest to add/update title although to show that information

Reply:
But these "coverage tests" are still host tests, correct? The title says that this file defines **all** workflows related to build and host tests (including the host tests with gcov enabled). The info about running the coverage analysis is still in the respective job name:
https://github.com/espressif/esp-protocols/pull/242/files#diff-2b92012e9e84fa80ea1bcff4a12777d87bcfef169dd05b9d9a39946e97d25261R88